core/store/src/lib.rs

```diff
@@ -289,27 +289,49 @@ pub fn read_with_cache<'a, T: BorshDeserialize + 'a>(
     Ok(None)
 }
 
-#[derive(Default, Debug)]
+#[derive(Debug, Clone)]
 pub struct StoreConfig {
     /// Attempted writes to the DB will fail. Doesn't require a `LOCK` file.
     pub read_only: bool,
     /// Re-export storage layer statistics as prometheus metrics.
     /// Minor performance impact is expected.
     pub enable_statistics: bool,
+    /// Maximum number of store files being opened simultaneously.
+    pub max_open_files: i32,
+    /// Cache size for ColState column.
+    pub col_state_cache_size: usize,
+}
+
+impl StoreConfig {
+    pub fn read_only() -> StoreConfig {
+        let mut config = StoreConfig::default();
+        config.read_only = true;
+        config
+    }
+}
+
+impl Default for StoreConfig {
+    fn default() -> StoreConfig {
```

Comment:
Perhaps it would be better to remove `Default` implementation and rename this method `read_write`?